AndelaOSP/wire
# #155394319 - Set errors in the store

#### What does this PR do?
Adds an Incident Error action creator that is dispatched when the loadIncidents thunk returns
an error.
#### Description of Task to be completed?
- We need to get the error in the state and show it to the user.
- When an error occurs, an appropriate action should be dispatched to set the error in the store to be displayed to the user.
#### Where should the reviewer start?
View file changes on PR.
#### How should this be manually tested?
Manually modify the mock api to return an error. For instance, change one of the functions in mockMiddleware.js from say:
```
fetchIncidents: (req, res) => {
    setTimeout(() => {
      res.send({ data: { incidents: getIncidents() }, status: 'success' });
    }, 2000);
  }
```
to:
```
fetchIncidents: (req, res) => {
    setTimeout(() => {
      res.status(404).send({ error: 'Not found', status: 'error' });
    }, 2000);
  }
```
Attempt to reload the dashboard. A custom toast message should appear with the error status code and error message. Also use redux dev tools to see if the incident error action creator was 
dispatched.
#### What are the relevant pivotal tracker stories?
#155394319
#### Screenshots:
<img width="1440" alt="screen shot 2018-03-01 at 11 36 59 am" src="https://user-images.githubusercontent.com/25130454/36834784-093cff5a-1d45-11e8-80e5-66d34e25b928.png">
<img width="1440" alt="screen shot 2018-03-04 at 3 09 43 pm" src="https://user-images.githubusercontent.com/25130454/36945382-3bf830c4-1fbe-11e8-9631-62c6a8f72f1d.png">
